src/snowflake/cli/_plugins/nativeapp/artifacts.py

```diff
@@ -240,11 +240,21 @@ class BundleMap:
     """
     Computes the mapping between project directory artifacts (aka source artifacts) to their deploy root location
     (aka destination artifact). This information is primarily used when bundling a native applications project.
+
+    :param project_root: The root directory of the project and base for all relative paths. Must be an absolute path.
+    :param deploy_root: The directory where artifacts should be copied to.
     """
 
     def __init__(self, *, project_root: Path, deploy_root: Path):
+        if not project_root.is_absolute():
+            raise ValueError("Project root must be an absolute path")
         self._project_root: Path = resolve_without_follow(project_root)
-        self._deploy_root: Path = resolve_without_follow(deploy_root)
+        abs_deploy_root = (
+            deploy_root
+            if deploy_root.is_absolute()
+            else self._project_root / deploy_root
+        )
```

Comment:
Is this necessary? I thought that `p1 / p2` already returns p2 if it's absolute.

Reply:
oh yeah that's true, I was just using that in an earlier PR 😅